Education_Pathways/frontend/src/components/ResultDisplay.jsx

```diff
@@ -27,30 +30,38 @@ class SearchResultDisplay extends Component{
     this.getData(this.state.input)
   }
 
+  handleSubmit2(event) {
+    event.preventDefault();
+    this.getData("hi")
+  }
+
   getData = (input) => {
-    axios.get(`https://assignment-1-starter-template.herokuapp.com/searchc?input=${input}`)
+    API.get(`/searchc?input=${input}`)
+    // axios.get(`https://assignment-1-starter-template.herokuapp.com/searchc?input=${input}`)
       .then(res => {
-        console.log(`it is ${res.status}`)
+        // console.log(`it is ${res.status}`)
         if (res.status === 200) {
           this.setState({results: []})
-          
+          console.log(res.data.length)
```

Comment:
This line will cause the application to crash when the user presses "Search" using a query that matches no courses. The API response `res` will be successful (i.e. `res.status` = 200) but it will not have any data (i.e. `res.data` = null) because no courses match the query. As a consequence, `res.data.length` will return a TypeError because `res.data` (which is null) has no attribute _**length**_.

Reply:
To fix this, I suggest replacing `if (res.data.length > 0){}` with `if (res.data != null){}`